surl/__init__.py

```diff
@@ -277,236 +285,49 @@ def get_config_from_cli(parser, auth_dir):
-            email,
-            permissions=permissions,
-            channels=channels,
-            allowed_stores=allowed_stores,
-            snaps=snaps,
-            store_env=store_env,
-        )
-
-    return root, discharge
-
-
+# Note that store_env only exists to make surl_metrics (and possibly others) happy
 def get_authorization_header(root, discharge, store_env=None):
-    """Bind root and discharge returning the authorization header."""
+    """Return the required authorization header, possibly binding the root and discharge"""
     root = Macaroon.deserialize(root)
     if discharge is not None:
         discharge = Macaroon.deserialize(discharge)
-        if discharge.location == CONSTANTS[store_env]["sso_location"]:
-            # U1 SSO macaroons auth
-            bound = root.prepare_for_request(discharge)
-            authorization = "Macaroon root={}, discharge={}".format(
-                root.serialize(), bound.serialize()
-            )
-            return {"Authorization": authorization}
-        else:
-            # to-be-deprecated: kept for backwards compatibility and
-            # existing Candid macaroons
-            return _get_bakery_auth_header(root, discharge)
+        bound = root.prepare_for_request(discharge)
+        authorization = (
+            f"macaroon root={root.serialize()}, discharge={bound.serialize()}"
+        )
+        return {"Authorization": authorization}
     else:
-        # snapstore-only macaroon auth
-        authorization = "Macaroon {}".format(root.serialize())
+        authorization = f"macaroon {root.serialize()}"
         return {"Authorization": authorization}
 
 
-def get_refreshed_discharge(discharge, store_env):
-    headers = DEFAULT_HEADERS.copy()
-    data = {"discharge_macaroon": discharge}
-    response = requests.request(
-        url="{}/api/v2/tokens/refresh".format(
-            CONSTANTS[store_env]["sso_base_url"]
-        ),
-        method="POST",
-        json=data,
-        headers=headers,
-    )
-    return response.json()["discharge_macaroon"]
+def get_client(web_login, store):
+    if web_login:
+        return StoreClient(
+            base_url=CONSTANTS[store]["sca_base_url"],
+            storage_base_url="https://storage.staging.snapcraftcontent.com",
+            endpoints=endpoints.SNAP_STORE,
+            user_agent=DEFAULT_HEADERS["User-Agent"],
+            application_name="surl",
+            environment_auth="CREDENTIALS",
+            ephemeral=True,
+        )
+    else:
+        return UbuntuOneStoreClient(
+            base_url=CONSTANTS[store]["sca_base_url"],
+            storage_base_url="https://storage.staging.snapcraftcontent.com",
+            auth_url=CONSTANTS[store]["sso_base_url"],
+            endpoints=endpoints.U1_SNAP_STORE,
+            user_agent=DEFAULT_HEADERS["User-Agent"],
+            application_name="surl",
+            environment_auth="CREDENTIALS",
+            ephemeral=True,
+        )
```

Comment:
I'm torn about the above two client constructors. In all likelyhood, the way you've done it is best. Just for sake of discussion though, do you think there is any merit in something like:

```python3
    common_args = dict(
        base_url=CONSTANTS[store]["sca_base_url"],
        storage_base_url="https://storage.staging.snapcraftcontent.com",
        user_agent=DEFAULT_HEADERS["User-Agent"],
        application_name="surl",
        environment_auth="CREDENTIALS",
        ephemeral=True,
    )
    if web_login:
        return StoreClient(
            endpoints=endpoints.SNAP_STORE,
            **common_args,
        )
    else:
        client = UbuntuOneStoreClient
            endpoints=endpoints.U1_SNAP_STORE,
            auth_url=CONSTANTS[store]["sso_base_url"],
            **common_args,
        )
```

Reply:
Hadn't really considered separating out the common arguments like that, I like it so I'll make the change.